### ISSUE TITLE
NtOS error when inputting a revolver into a custom vending machine (And potentially other items)

### ISSUE THREAD

**Issue by Fastmapler** [tgstation/tgstation#63307]
## Round ID: 
[174423](https://scrubby.melonmesa.com/round/174423)

## Reproduction:

When setting up a custom vending machine, I tagged a 357 revolver for 500cr and put it into the machine. However, trying to access the machine resulted in a NtOS bug as follows. Trying to re-access the machine yielded the same error, and another player tried to access the vending but also saw the error. On a local build I tried this again and got the same issue. Also seems to happen when I inserted a tagged wrench.

TypeError: Unable to get property 'amount' of undefined or null reference
   at s (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:1531102)
   at Ie (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179474)
   at Anonymous function (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179742)
   at Le (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179720)
   at De (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:180768)
   at Te (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:180607)
   at Le (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179584)
   at Te (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:180573)
   at Le (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179584)
   at Anonymous function (http://127.0.0.1:61208/tmp25172/tgui.bundle.js:1:179742)
User Agent: Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.2; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; .NET CLR 2.0.50727; .NET CLR 3.0.30729; .NET CLR 3.5.30729)
State: {"ckey":"fastmapler","interface":"Vending","window":{"key":"[0x2006310]-main","size":null,"fancy":1,"locked":0}}

In short:
1. Build a Custom Vendor using a Custom Vendor machine board and resupply canister
2. Link the vending machine to you by using your ID card on it
3. Obtain a 357 revolver, and use a price tagger on it to set a price
4. Insert the revolver into the machine, and attempt to open the vending machine's interface

**Comment by TheBonded** [comment]
#59800 seems to have not been fixed

**Comment by jlsnow301** [comment]
Vending machine code is like a hydra of bugs

**Comment by timothymtorres** [comment]
Could be related to #50569

> Vending machine code is like a hydra of bugs

Fix one bug and 7 more appear.  It’s the bugpocalypse!
